### PR TITLE
[4.0] [mod_articles_category] Always render module layout

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes

Per discussion in #20811, module layouts should always be rendered even if no items are available.

### Testing Instructions

Create `mod_articles_category` override which always displays some custom message/code.
Publish `Articles - Category` module. Configure filtering options so that no articles are displayed.
View the module in frontend.

### Expected result

Custom message shown.

### Actual result

Custom message not shown.

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by brianteeman** [comment]
What happens if there is no custom message? Do you end up with an empty space? If so then I stick by my comments on that discussion that this is not desirable as you "may" end up with empty columns

**Comment by SharkyKZ** [comment]
You end up with an empty column anyways. Read discussion in #20811 please (you were involved in it). There's also PR to for excluding empty modules in templates, see #19416.

File: modules/mod_articles_category/mod_articles_category.php [hunk 1]
```diff
@@ -59,41 +59,33 @@
...
+				$params->get('month_year_format', 'F Y'),
+				$params->get('date_grouping_field', 'created')
+			);
+			break;
+		case 'author' :
+		case 'category_title' :
+			$list = ArticlesCategoryHelper::groupBy($list, $article_grouping, $article_grouping_direction);
+			break;
+		case 'tags' :
+			$list = ModArticlesCategoryHelper::groupByTags($list, $article_grouping_direction);
```
Comment (Quy) [inline modules/mod_articles_category/mod_articles_category.php]:
```suggestion
			$list = ArticlesCategoryHelper::groupByTags($list, $article_grouping_direction);
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on afc8cfaaaba929d1c6ecf370925fb303e3d39fc4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24956">issues.joomla.org/tracker/joomla-cms/24956</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on afc8cfaaaba929d1c6ecf370925fb303e3d39fc4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24956">issues.joomla.org/tracker/joomla-cms/24956</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by wilsonge** [comment]
> What happens if there is no custom message? Do you end up with an empty space? If so then I stick by my comments on that discussion that this is not desirable as you "may" end up with empty columns

https://github.com/joomla/joomla-cms/pull/20811#issuecomment-399020843 you mentioned here we already have that behaviour so this change makes sense to allow customisation. However clearly we can probably improve this behaviour - but it's beyond the scope of this PR

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**